### PR TITLE
feat: add yearly schedule to presets

### PR DESCRIPTION
**Pull request by HonkingGoose**
## Changes:

- Add yearly schedule to preset schedule

## Context:

I saw that a user wanted to configure Renovate to open a PR once a year, and had some difficulty to figure out how to do this.
I went digging in the source code, and it seems easy to add a yearly schedule.

This is basically a _copy/paste_ of the quarterly schedule, except the schedule runs once every 12 months, instead of every 3 months.
Plus I changed the description to match the schedule.

Inspired by the discussion at https://github.com/renovatebot/renovate/discussions/8226.

**Questions:**

- What happens if a user enables the schedule on February 29th (leap day), does the bot still open a PR on March 1st?
- When does the schedule first run (I suppose it runs on first day of the month after it's enabled, say you enable the config 8th of January, the bot would run on the first of February.)

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

The https://docs.renovatebot.com/presets-schedule/ configuration documentation is auto-generated based on the file that I'm changing.

I don't think we need more documentation?

## How I've tested my work (please tick one)

I have verified these changes via:

- [x] Code inspection only, or
- [ ] Newly added unit tests, or
- [ ] Unit tests + ran on a real repository

I don't know if we need a new unit test to verify the behavior.

### CODE REVIEW

**Review by viceice — DISMISSED** [review]
I think it's a very bad idea to only update yearly. But otherwise LGTM

**Comment by rarkins** [comment]
Are we sure this works as intended? Hard to test in practice

**Comment by HonkingGoose** [comment]
@rarkins Because it's only once every 12 months, this is not easy to verify.

Also what happens if the bot is restricted via another schedule, and the bot misses the timeslot for the 1st of the month? Does the bot then retry, or give any indication that it's missed it's timeslot? I think we currently just warn users of the bot to not restrict the bot schedule too much. So that might be another reason to not nudge to using only once every 12 months.

I was just wondering, it seems really "easy to add", why isn't this already in the presets...
But things sometimes are only easy on the surface and hard in practice... 😄 

> Are we sure this works as intended? Hard to test in practice

I agree this is hard to test.

We allow a once every three months schedule in the preset, I think that's also hard to test in practice, but of-course once every 12 months is even harder to test... 😄 How did/are we verify(ing) the 3 months schedule then? Did we just test the preset manually before adding it to the presets?

I just wanted to open the discussion on this.
Don't feel pressured to merge this if you have doubts!

**Comment by JamieMagee** [comment]
I added the quarterly preset. I verified by setting the schedule in my configuration, and validating that renovate is able to parse it, and generate the next dates correctly.

I just did the same for this preset, and it generates the next dates correctly as well. I think this is as close to a full validation as we'll get. However, you're right, that if users have configured a schedule + renovate actually running that don't line up, then they'll never get any PRs.

**Comment by viceice** [comment]
I think we should add some description to not use schedules less than montly / quaterly, because of long outstanding critical security updates which can /will be missing then.

File: lib/config/presets/internal/schedule.ts [hunk 1]
```diff
@@ -21,6 +21,10 @@ export const presets: Record<string, Preset> = {
     description: 'Schedule quarterly',
     schedule: ['every 3 months on the first day of the month'],
   },
+  yearly: {
+    description: 'Schedule once a year',
```
Comment (HonkingGoose) [inline lib/config/presets/internal/schedule.ts:25]:
```suggestion
    description: 'Schedule once a year (think before you use this preset, you'll miss critical security updates)',
```

Comment (rarkins) [reply lib/config/presets/internal/schedule.ts:25]: I think so, but maybe just use `(not recommended)` ?

**Review by rarkins — APPROVED** [review]

**Pull request merged**